### ISSUE TITLE
HDF5 output for HEDMInstrument.pull_spots() is not working

### ISSUE THREAD

**Issue by joelvbernier** [HEXRD/hexrd#169]
This seems to be something that got broken moving from Python2.7 to Python3.  I am seeing a `TypeError` coming from the utility func `instrument.unwrap_dict_to_h5()`.  Looks like a unicode string type might be responsible?  Can you help with this @donald-e-boyce or @psavery?

```
instr.pull_spots(plane_data, grain_parameters, ims_dict, tth_tol=0.3, eta_tol=4.0, ome_tol=4.0, threshold=25, dirname="./", filename="grain_00000.hdf5", output_format='hdf5')
Traceback (most recent call last):

  File "<ipython-input-83-910ad91e77ea>", line 1, in <module>
    instr.pull_spots(plane_data, grain_parameters, ims_dict, tth_tol=0.3, eta_tol=4.0, ome_tol=4.0, threshold=25, dirname="./", filename="grain_00000.hdf5", output_format='hdf5')

  File "c:\users\berni\onedrive\documents\github\hexrd\hexrd\instrument.py", line 1447, in pull_spots
    writer = GrainDataWriter_h5(

  File "c:\users\berni\onedrive\documents\github\hexrd\hexrd\instrument.py", line 3126, in __init__
    unwrap_dict_to_h5(self.instr_grp, icfg, asattr=use_attr)

  File "c:\users\berni\onedrive\documents\github\hexrd\hexrd\instrument.py", line 3235, in unwrap_dict_to_h5
    grp.create_dataset(key, data=np.atleast_1d(item))

  File "C:\Users\berni\miniconda3\envs\hexrd_0.8\lib\site-packages\h5py\_hl\group.py", line 136, in create_dataset
    dsid = dataset.make_new_dset(self, shape, dtype, data, **kwds)

  File "C:\Users\berni\miniconda3\envs\hexrd_0.8\lib\site-packages\h5py\_hl\dataset.py", line 118, in make_new_dset
    tid = h5t.py_create(dtype, logical=1)

  File "h5py\h5t.pyx", line 1634, in h5py.h5t.py_create

  File "h5py\h5t.pyx", line 1656, in h5py.h5t.py_create

  File "h5py\h5t.pyx", line 1717, in h5py.h5t.py_create

TypeError: No conversion path for dtype: dtype('<U10')
```

**Comment by joelvbernier** [comment]
Looks like it is related to [this](https://docs.h5py.org/en/stable/strings.html#what-about-numpy-s-u-type).  Catch and convert in-line?

**Comment by psavery** [comment]
Yeah, it looks like `<U*` is numpy's wide character type. h5py says that hdf5 does not support wide characters, so they don't either.

Maybe right before the dataset gets created, we can add a check like this?

```python
if item.dtype.type == np.str_:
    item = item.astype(h5py.special_dtype(vlen=str))
```

Or if another type is better, we can convert it to that as well.

Let me know if that works, @joelvbernier. If not, maybe we can get an example to try?

**Comment by joelvbernier** [comment]
I think I fixed it in #171 -- was able to dump the HDF5 tables again.  Closing :-)
